### PR TITLE
Add -ftime-profile documentation

### PR DESCRIPTION
**Pull request by jorisv**

### CODE REVIEW

**Review by fabinsch — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -261,3 +261,36 @@ Run the CI and follow the output. Note: the option `continue-on-error: True` can
   uses: lhotari/action-upterm@v1
 ```
 Consider using the action with `limit-access-to-actor: true`, to limit access to your account.
+
+
+## Profile C++ compile time
+
+When doing heavy template meta-programming in C++ it can be useful to analyze what part of the code is taking a long time to compile.
+
+[clang](https://clang.llvm.org) allow to profile the compilation time. To activate this function, add `-ftime-trace` option while building.
```
Comment (stephane-caron) [inline README.md:270]:
```suggestion
[clang](https://clang.llvm.org) allows to profile the compilation time. To activate this function, add the `-ftime-trace` option while building.
```

File: README.md [hunk 2]
```diff
@@ -261,3 +261,36 @@ Run the CI and follow the output. Note: the option `continue-on-error: True` can
...
+
+On CMake project, you can do this with the following command line:
```
Comment (stephane-caron) [inline README.md:272]:
```suggestion
In a CMake project, you can do this with the following command line:
```

File: README.md [hunk 3]
```diff
@@ -261,3 +261,36 @@ Run the CI and follow the output. Note: the option `continue-on-error: True` can
...
+cmake .. -DCMAKE_CXX_FLAGS="-ftime-trace"
+```
+
+Each .cpp will then produce a .json file. To find them you can use the following command line:
+
+```bash
+find . -iname "*.json"
+```
+
+Then, you can open the file with Chromium tracing tool. Open the `about:tracing` URL in Chromium and load the .json. You will have the following display:
```
Comment (stephane-caron) [inline README.md:284]:
```suggestion
Then, you can open the file with the [Chromium tracing tool](https://www.chromium.org/developers/how-tos/trace-event-profiling-tool/). Open the `about:tracing` URL in Chromium and load the .json. You will have the following display:
```

File: README.md [hunk 4]
```diff
@@ -261,3 +261,36 @@ Run the CI and follow the output. Note: the option `continue-on-error: True` can
...
+
+![Compile time profile displayed with Chromimum](./resources/cpp-compilation-time-profile.png)
+
+### Note for GNU/Linux user
```
Comment (stephane-caron) [inline README.md:288]:
```suggestion
### Note for GNU/Linux users
```

GNU/Linux :heart:

**Review by stephane-caron — APPROVED** [review]
Awesome, thanks @jorisv :+1: 

Minor suggestions in comments.

**Comment by jorisv** [comment]
@stephane-caron I have apply all your suggestions.

**Review by stephane-caron — APPROVED** [review]

**Review by jcarpent — APPROVED** [review]

**Pull request merged**